From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Driver core: marker to keep driver_data from being used

We are trying to catch all users of this field...

---
 include/linux/device.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -388,7 +388,7 @@ struct device {
 	struct bus_type	*bus;		/* type of bus device is on */
 	struct device_driver *driver;	/* which driver has allocated this
 					   device */
-	void		*driver_data;	/* data private to the driver */
+	void		*dd_do_not_touch;	/* data private to the driver */
 	void		*platform_data;	/* Platform specific data, device
 					   core doesn't touch it */
 	struct dev_pm_info	power;
@@ -456,12 +456,12 @@ static inline void set_dev_node(struct d
 
 static inline void *dev_get_drvdata(const struct device *dev)
 {
-	return dev->driver_data;
+	return dev->dd_do_not_touch;
 }
 
 static inline void dev_set_drvdata(struct device *dev, void *data)
 {
-	dev->driver_data = data;
+	dev->dd_do_not_touch = data;
 }
 
 static inline unsigned int dev_get_uevent_suppress(const struct device *dev)
